modules.d/06hpsa/parse-hpsa.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin.bash
+### Laurence Oberman loberman@redhat.com
+### parse-hpsa.sh
+### Parses the rd.hpsa=x tp get the host number
+### Using rdloaddriver=hpsa will enforce hpsa becoming scsi0
+
+for p in $(getargs rd.hpsa=); do
+(
+     echo "echo 1 > /sys/class/scsi_host/host$p/rescan" > /sbin/hpsa_scan.sh
+    _do_hpsa=1
+)
+done
+
+### Standard way to call the script from udev
+/sbin/initqueue --settled --unique --onetime /bin/sh /sbin/hpsa.sh
```

Comment:
That script is built based on the SCSI host# as passed in the kernel command line. It would need to be created with the correct SCSI host#.
This is why it is dynamically created